### ISSUE TITLE
Error on load extension

### ISSUE THREAD

**Issue by obayit** [danielfalk/smart-open.nvim#5]
The error:
```stack
Error detected while processing :source (no file):                                                                                            
E5108: Error executing lua ...im/site/pack/packer/start/sqlite.lua/lua/sqlite/defs.lua:55: libsqlite3.so: cannot open shared object file: No such file 
or directory
stack traceback:
        [builtin#203]: at 0x55f830986100
        ...im/site/pack/packer/start/sqlite.lua/lua/sqlite/defs.lua:55: in main chunk
        [C]: in function 'require'
        ...m/site/pack/packer/start/sqlite.lua/lua/sqlite/utils.lua:252: in function '__index'
        ...nvim/site/pack/packer/start/sqlite.lua/lua/sqlite/db.lua:662: in main chunk
        [C]: in function 'require'
        ...im/site/pack/packer/start/sqlite.lua/lua/sqlite/init.lua:66: in function '__index'
        ...n.nvim/lua/telescope/_extensions/smart_open/dbclient.lua:27: in function 'new'
        ...smart-open.nvim/lua/telescope/_extensions/smart_open.lua:52: in function 'setup'
        .../start/telescope.nvim/lua/telescope/_extensions/init.lua:64: in function 'load_extension'
        [string ":source (no file)"]:11: in main chunk
```

How the plugin is installed:
```lua
    use {
        "danielfalk/smart-open.nvim",
        requires = {
            { "tami5/sqlite.lua" },
            { 'kyazdani42/nvim-web-devicons', opt = true },
        }
    }

```

relevant part of my `after/plugin/telescope.lua`
```lua
telescope.load_extension("smart_open")
vim.keymap.set("n", "<Leader><Leader>", telescope.extensions.smart_open.smart_open)

```

**Comment by danielfalk** [comment]
Hmm.  I have no idea, but I think this is a sqlite.lua issue, and possibly related to your platform.  What system are you running?

**Comment by danielfalk** [comment]
Also, do you have sqlite3 installed locally? There are instructions on it here: [https://github.com/kkharji/sqlite.lua#readme](https://github.com/kkharji/sqlite.lua#readme) (The repository seems to have been renamed recently)

It looks like I should add a note to the installation docs to ensure that people have sqlite3 installed, but please let me know if the above was your problem!

**Comment by danielfalk** [comment]
I take it you figured this out, since I saw a different report from you that you would have had to get this working to have made.  Feel free to reopen if you are still experiencing.

**Comment by obayit** [comment]
@danielfalk  sorry I didn't see your comments, it works on one machine but doesn't in the other! they are both ubuntu 20.04
But yeah it is an issue on my end.

**Comment by obayit** [comment]
yes it worked after I installed sqlite3
thanks
@danielfalk

**Comment by Jerakin** [comment]
I am getting this error too.

I am on windows, I am not that deep into the echosystem (yet) so the instructions.
> **Windows**
> [Download precompiled](https://www.sqlite.org/download.html) and set `let g:sqlite_clib_path = path/to/sqlite3.dll` (note: /)

Is not enough for me here. I have downloaded the precompiled files, but I still get the error `defs.lua:57: cannot load module 'libsqlite3': The specified module could not be found.`. I tried to add the folder both to `PATH` and to a env-var called `sqlite_clib_path` but neither worked (tried to change the pathsep too).

Is there something I am missing? What does the last part mean `let g:...`?

Specs: 
Windows 10
NVIM v0.9.5 (Used [kickstart](https://github.com/nvim-lua/kickstart.nvim) to get up and running.
